File: 307. Range Sum Query - Mutable.py
class NumArray(object):
    def __init__(self, nums):
        """
        initialize your data structure here.
        :type nums: List[int]
        """
        self.nums=nums

    def update(self, i, val):
        """
        :type i: int
        :type val: int
        :rtype: int
        """
        self.nums[i]=val

    def sumRange(self, i, j):
        """
        sum of elements nums[i..j], inclusive.
        :type i: int
        :type j: int
        :rtype: int
        """
        return sum(self.nums[i:j+1])


# Your NumArray object will be instantiated and called as such:
# numArray = NumArray(nums)
# numArray.sumRange(0, 1)
# numArray.update(1, 10)
# numArray.sumRange(1, 2)

File: 307. Range Sum Query - Mutable.cpp
class NumArray {
private:
    vector<int> numlist;
public:
    NumArray(vector<int> &nums) {
        numlist = nums;
    }

    void update(int i, int val) {
        numlist[i] = val;
    }

    int sumRange(int i, int j) {
        int res = 0;
        for (int k = i; k <= j; k++)
            res += numlist[k];
        return res;
    }
};


// Your NumArray object will be instantiated and called as such:
// NumArray numArray(nums);
// numArray.sumRange(0, 1);
// numArray.update(1, 10);
// numArray.sumRange(1, 2);
